src/components/share-button/share-button.component.ts

```diff
@@ -105,24 +73,25 @@ export class ShareButtonComponent implements AfterViewInit {
                 this.popUpClosed.emit(this.button.provider);
             }
         }, 200);
-    }
+    };
 
+    shareButton() {
+        this.btn = this.renderer.createElement(this.self, 'button');
+        this.renderer.setElementProperty(this.btn, 'innerHTML', this.button.template);
+        this.renderer.setElementAttribute(this.btn, 'class', this.button.classes);
+        this.renderer.setElementProperty(this.btn, 'onclick', this.share);
+    }
 
-    nFormatter(num, digits) {
-        let si = [
-            { value: 1E18, symbol: 'E' },
-            { value: 1E15, symbol: 'P' },
-            { value: 1E12, symbol: 'T' },
-            { value: 1E9, symbol: 'G' },
-            { value: 1E6, symbol: 'M' },
-            { value: 1E3, symbol: 'K' }
-        ], rx = /\.0+$|(\.[0-9]*[1-9])0+$/, i;
-        for (i = 0; i < si.length; i++) {
-            if (num >= si[i].value) {
-                return (num / si[i].value).toFixed(digits).replace(rx, '$1') + si[i].symbol;
-            }
-        }
-        return num.toFixed(digits).replace(rx, '$1');
+    shareCount() {
+        this.sbService.count(this.button.provider, this.url)
+            .subscribe(shareCount => {
+                if (shareCount) {
+                    let counter = this.renderer.createElement(this.self, 'span');
```

Comment:
Following the **"avoid DOM manipulation"** comment, this should be in the template of `share-button`,  and controlled by a `ngIf` on public field `shareCount`